src/components/LocaleSelector.vue

```diff
@@ -37,12 +40,18 @@ export default {
   },
   data() {
     return {
-      langs: Object.keys(locales),
+      locales,
     };
   },
   methods: {
-    getLanguageName(lang) {
-      return locales[lang];
+    updateRouter({ target: { value: currentLocale } }) {
+      this.$i18n.locale = currentLocale;
+      this.$router.push({
```

Comment:
In addition to updating the URL, do you think we should also add some basic persistence logic so that this choice is saved across sessions in the same manner that we persist the Swift vs ObjC choice for certain APIs?

As a user, I think it would be nice if I didn't always have to use the toggle to choose an alternate locale preference whenever I close/re-open a new browser tab.

Reply:
Yes, I have it in mind. This is part of rdar://104924748. Out of scope for this one